requirements-dev.txt

```diff
@@ -3,6 +3,7 @@ pre-commit
 pytest==7.1.3
 pytest-asyncio==0.19.0
 pytest-timeout==2.1.0 # used to timeout tests
+pytest-xdist[psutil]==2.5.0  # xdist runs tests in parallel, psutil for better cpu no. detection
```

Comment:
I re-checked and psutil is not pure-python so that installation may not be available on all setups. Maybe skip the requirement here and instead manually install it in the test workflow? that way we avoid headaches for contribs trying to install the dev reqs